extralit-server/src/extralit_server/jobs/ocr_jobs.py

```diff
@@ -12,3 +12,243 @@
+        # current_job.save_meta()
+        raise
+
+
+def _call_marker_layout_detection(pdf_path: str, pages: Optional[str] = None) -> dict[str, Any]:
+    """
+    Call Marker's layout detection API using the standard PdfConverter.
+
+    Args:
+        pdf_path: Path to the PDF file
+        pages: Optional list of page numbers to process (0-indexed)
+
+    Returns:
+        Marker's layout detection results with block structure
+    """
+    # Basic input validation
+    if not pdf_path:
+        raise ValueError("PDF path cannot be empty")
+
+    pdf_file = Path(pdf_path)
+    if not pdf_file.exists():
+        raise FileNotFoundError(f"PDF file does not exist: {pdf_path}")
+
+    if pdf_file.suffix.lower() != ".pdf":
+        raise ValueError(f"File is not a PDF: {pdf_path}")
+
+    try:
+        # Create optimized configuration for layout detection
+        config_dict = {
+            "output_format": "json",
+            "parallel_factor": 1,
+        }
+
+        if pages is not None:
+            config_dict["page_range"] = pages
+
+        config = ConfigParser(config_dict)
+        model_dict = create_model_dict()
+
+        converter = PdfConverter(
+            config=config.generate_config_dict(),
+            artifact_dict=model_dict,
+        )
+
+        # Convert PDF - this will return a Document object with detected layout
+        result: "MarkdownOutput | JSONOutput" = converter(pdf_path)  # noqa: UP037
+        print(type(result))
+        pprint(result.model_dump())
+
+        # Extract layout information from the result
+        # The result should have metadata and blocks that we can process
+        layout_data = {"pages": []}
+
+        if hasattr(result, "pages") and result.pages:
+            for page_idx, page in enumerate(result.pages):
+                page_data = {"page": page_idx, "blocks": []}
+
+                # Extract blocks from the page
+                if hasattr(page, "blocks") and page.blocks:
+                    for block in page.blocks:
+                        if hasattr(block, "block_type") and hasattr(block, "bbox"):
+                            block_data = {
+                                "type": str(block.block_type).lower(),
+                                "bbox": list(block.bbox) if block.bbox else [],
+                                "id": str(getattr(block, "id", "")),
+                                "score": getattr(block, "confidence", 1.0),
+                            }
+
+                            # Add content based on block type
+                            if hasattr(block, "content"):
+                                block_data["content"] = str(block.content)
+                            elif hasattr(block, "text"):
+                                block_data["content"] = str(block.text)
+
+                            page_data["blocks"].append(block_data)
+
+                layout_data["pages"].append(page_data)
+
+        return layout_data
+
```

Comment:
Parsing the layout json from marker output is currently broken, since `results` is being returned as a `MarkdownOutput` type, not `JSONOutput`. You may have to dig a bit to see how to do this, since `marker` don't really have documentations for it